CIP-0106/README.md

```diff
@@ -0,0 +1,173 @@
+---
+CIP: 106
+Title: Web-Wallet Bridge - Mutlisig wallets
+Status: Proposed
+Category: Wallets
+Authors: 
+  - Leo
+Implementors: NA
+Discussions:
+Created: 2023-10-12
+License: CC-BY-4.0
+---
+
+## Abstract
+
+This document describes a CIP-30 extension allowing webpages (i.e. dApps) to interface with Cardano Multisig-wallets. This document is a work in progress and is not yet finalized. It is expected to be updated as the ecosystem evolves. 
+
+## Motivation: why is this CIP necessary?
+
+In order to facilitate future dApp development, we will need a way for dApps to communicate with multisig wallets, given the unique complexities of native script based addresses. Special provisions need to be made to make the connector compatible with them.  
+
+Specifically, apps building transactions need to be able to get the following information from the wallet:
+- Script descriptor 
+- Script Requirements list 
+- Collateral donator (since native script based addresses cannot provide collateral for transactions)
+
+Additionally, apps need to be able to submit a transaction to the wallet for signing in an asynchronous manner, as gathering of signatures can take a long time and each wallet provider will have its own way of handling this process. 
+
+Finally, the signTx() and signData() endpoints will have to be disabled when using this extension since they are not compatible with native script based addresses.
+
+### Rationale for the required data
+
+- Script descriptor:
+	Any transaction consuming a UTxO from a native script based address must attach the corresponding script. 
+- Script Requirements list:
+	-- dApps need to know the number of signers to calculate the fees correctly.
+	-- Including the correct "required signers" brings UX improvements.
+	-- dApps need to know if the script has any before-after requirements to set the correct validBefore and TTL values.
+- Collateral donator
+
+Native script based addresses cannot provide collateral for Plutus transactions 
+
+## Specification
+
+### Data Types
+
+#### KeyHash
+
+A hex-encoded string of the corresponding bytes. This represents the hash of the public key used to sign transactions.
+
+```ts
+type KeyHash = String
+```
+
+#### ScriptRequirements
+
+```ts
+type ScriptRequirementsCode = {
+    Signer: 1,
+    Before: 2,
+    After: 3,
+}
+type ScriptRequirement = {
+    code: ScriptRequirementsCode,
+    value: KeyHash|number,
+}
+```
+
+### Aditional Error Types
+
+#### CompletedTxError
+
+```ts
+CompletedTxErrorCode = {
+	NotFound: 1,
+	NotReady: 2
+}
+```
+
+* NotFound - The transaction with the given id was not found.
+* NotReady - The transaction with the given id is not ready yet. 
+
+## Motivation: why is this CIP necessary?
```

Comment:
duplicate section

Reply:
@Ryun1 not really a duplicate section but @leo42 this is the same problem as I had to fix in https://github.com/cardano-foundation/CIPs/pull/798#discussion_r1586188780 where "Rationale for the required data" in the Motivation... really just a definition of specification terms... seemed to be a separate section.

Since your material is in exactly the same form here, it can be fixed exactly the same way as in https://github.com/cardano-foundation/CIPs/pull/798/commits/25b463c99f77dc48d4f8034a3e06e6bfd9fa8323.

Reply:
@leo42 - https://github.com/cardano-foundation/CIPs/pull/617/commits/9d6a57f339fe1ae3c65ed418a716f1495d8ad38c doesn't fix this problem (an apparently duplicated Rationale section).  Removing that title — and joining the "required data" terms with their definitions, as suggested above — _would_.

Reply:
Fixed by https://github.com/cardano-foundation/CIPs/pull/617/commits/ce033b7d29946956417015b9efe8f3af14d06576 ... although FYI @leo42 it was not necessary to remove these "required data" definitions (they could simply be attached to the terms they define, as we did in https://github.com/cardano-foundation/CIPs/commit/25b463c99f77dc48d4f8034a3e06e6bfd9fa8323).

Reply:
Good suggestion, sorry I have gotten lost among the several branches of this conversation

Reply:
understood @leo42 & fully resolved in https://github.com/cardano-foundation/CIPs/pull/617/files/483b8c87c3872ac44e5f054911dcbea4877969de..1d13e7073301793d2a7e3f54aeea3709f931d27e. 😎